resqpy/olio/relperm.py

```diff
@@ -0,0 +1,316 @@
+      x_new = saturation
+      assert x.min() <= x_new <= x.max(), 'saturation value is outside the interpolation range'
+      y_new = f(x_new)
+      return saturation, y_new.item()
+
+   def df_to_text(self, filepath, filename):
+      """Creates text file from a dataframe of relative permeability and capillary pressure data.
+
+        arguments:
+            filepath (str): location where new text file is written to
+            filename (str): name of the new text file
+        returns:
+            tuple of float, the first element is the saturation and the second element is the interpolated value
+           """
+      df = self.df.copy()
+      ascii_file = os.path.join(filepath, filename + '.dat')
+      df.columns = map(str.upper, df.columns)
+      if {'KRW', 'KRO'}.issubset(set(df.columns)):
+         if self.low_sal:
+            table_name_keyword = 'WOTABLE (LOW_SAL)\n'
+         else:
+            table_name_keyword = 'WOTABLE\n'
+         df.rename(columns = {'SW': 'SW', 'KRW': 'KRW', 'KRO': 'KROW', 'PC': 'PCWO'}, inplace = True)
+      elif {'KRG', 'KRO'}.issubset(set(df.columns)):
+         table_name_keyword = 'GOTABLE\n'
+         df.rename(columns = {'SG': 'SG', 'KRG': 'KRG', 'KRO': 'KROG', 'PC': 'PCGO'}, inplace = True)
+      elif {'KRW', 'KRW'}.issubset(set(df.columns)):
+         table_name_keyword = 'GWTABLE\n'
+         df.rename(columns = {'SG': 'SG', 'KRG': 'KRG', 'KRW': 'KRWG', 'PC': 'PCGW'}, inplace = True)
+      else:
+         raise Exception('incorrect rel. perm. column combination encountered')
+
+      if filename + '.dat' not in os.listdir(filepath):
+         with open(ascii_file, 'w') as f:
+            f.write(table_name_keyword)
+            df_str = df.to_string(na_rep = '', index = False)
+            f.write(df_str)
+            f.close()
+            print(f'Created new DAT file: {filename} at {filepath}')
+      else:
+         with open(ascii_file, 'a') as f:
+            f.write('\n\n')
+            f.write(table_name_keyword)
+            df_str = df.to_string(na_rep = '', index = False)
+            f.write(df_str)
+            f.close()
+            print(f'Appended to DAT file: {filename} at {filepath}')
+
+   def write_hdf5_and_create_xml(self):
+      """Write relative permeability table data to hdf5 file and create xml for RESQML
+        objects to represent dataframe."""
+
+      super().write_hdf5_and_create_xml()
+      # note: time series xml must be created before calling this method
+      mesh_root = self.mesh.root
+      # create an xml of extra metadata to indicate that this is a relative permeability table
+      rqet.create_metadata_xml(mesh_root, {'relperm_table': 'true'})
+
+
+def text_to_relperm_dict(filepath):
+   """
+    Returns a dictionary that contains dataframes with relative permeability and capillary pressure data and
+    phase combinations.
+
+    arguments:
+    filepath (str): relative or full path of the text file to be processed
```

Comment:
I think the indentation in this docstring needs to be fixed: "arguments" would be one indent, "filepath" would be two indents.

With resqpy's unusual 3-space indentation this might be difficult depending on your editor settings!

Reply:
I think I fixed the docstring issues based on your previous advice....maybe